odl/test/solvers/functional/default_functionals_test.py

```diff
@@ -505,6 +505,23 @@ def test_moreau_envelope_l2_sq(space, sigma):
     assert all_almost_equal(smoothed_l2_sq.gradient(x),
                             x * 2 / (1 + 2 * sigma))
 
+def test_weighted_proximal():
+    """Test for the weighted proximal of a SeparableSum functional."""
+
+    space = odl.space.rn(2)
+
+    l1 = odl.solvers.L1Norm(space)
+    l2 = odl.solvers.L2Norm(space)
+
+    x = space.one()
+    y = space.one()
+
+    func = odl.solvers.SeparableSum(l1, l2)
+
+    sigma = [0.5, 1.0]
+
+    prox = func.proximal(sigma)([x, y])
+    assert all_almost_equal(prox, [l1.proximal(0.5)(x), l2.proximal(1.0)(y)])
```

Comment:
this violates [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).  Either do simply

```python
prox = func.proximal([0.5, 1.0])([x, y])
expected = [l1.proximal(0.5)(x), l2.proximal(1.0)(y)])
assert all_almost_equal(prox, expected)       
```

or go for

```python
x = func.domain.one()
sigma = [0.5, 1.0]

prox = func.proximal(sigma)(x)
expected = [l1.proximal(sigma[0])(x[0]), l2.proximal(sigma[1])(x[1])])
assert all_almost_equal(prox, expected)       
```